documentation/ingestion_server/guides/migrate.md

```diff
@@ -0,0 +1,245 @@
+API. As such they do not need any migration process. Examples:
+
+- addition of new fields or subfields
+- removal of fields that are not referenced or used by the API
+- changing the type to another compatible type (like `text` &harr; `keyword` )
+
+For API-free changes, we deploy the ingestion server and let a data-refresh to
+occur. The indexes will be updated to the new schema without manual intervention
+and will be made available to the API.
+
+### API-involved
+
+These changes are modifications to fields that already are in use by the API and
+involve code changes in both the ingestion server and the API. Examples:
+
+- removal of a field
+- changing the type to an incompatible type
+- renaming of a field
+
+Such kinds of changes need us to precisely deploy the API when the new index is
+promoted because of these reasons:
+
+- If we deploy a little late, the old field the API wants will disappear.
+- If we deploy a little early, the new field the API wants will not be present.
+
+This runbook documents guidelines and processes for API-involved migrations.
+
+Our goal is to break down an API-involved change into multiple small, atomic
+changes with each step affecting at most one of the ingestion server or the API
+and ensuring that the API and ES remain compatible throughout the process.
+
+## Pull request guidelines
+
+A change that involves modification to the ES index as well as its usage in the
+API requires at least three steps, each associated with exactly one PR that
+modifies exactly one of the ingestion server or the API to allow them to be
+deployed independently.
+
+1. Change the ES index mapping in the ingestion server. Ensure that the change
+   is purely additive, keeping the old fields unchanged and creating new fields
+   that contain the data the API will need.
+
+   This PR should only make changes within the `ingestion_server/` directory.
+
+2. Change the ES fields referenced by the API to use the new fields added in the
+   previous step. Ensure that the old fields become unreferenced.
+
+   The PR should only make changes within the `api/` directory.
+
+3. Change the ES index mapping in the ingestion server to remove the old,
+   now-unreferenced fields.
+
+   This PR should only make changes within the `ingestion_server/` directory.
+
+```{tip}
+Get the PRs reviewed in advance so that the entire process has been vetted by
+the team and there are no surprises or delays when the plans have been set into
+motion.
+```
+
+```{caution}
+Each PR in the chain should branch from, and point to, its predecessor in the
+chain so that CI continues to pass for each PR.
+```
+
+### Example
+
+Assume we have a field `foo` with type `text` in the index. It has a subfield
+`keyword` with type `keyword`. The API uses `foo.keyword` for all purposes.
+
+One PR that does these 3 things would be an API-involved change. So we split
+them into 3 PRs.
+
+1. Changing `foo` to type `keyword` would be an API-free change because it is a
```

Comment:
_I think_ this is an API-free change because we are not using the `foo` field as a `keyword` field in the API, right? Or was it like only during our latest ES changes (we never used the `source` field in the API, only the `source.keyword`)

Reply:
Building on [this response](https://github.com/WordPress/openverse/pull/3321#discussion_r1385212742), a field with type `keyword` does not need the `.keyword` suffix, that suffix was a result of us using the a nested field named `keyword` inside the `source` field.